core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -951,6 +965,36 @@ private[spark] class BlockManager(
     None
   }
 
+  /**
+   * Reads the block from the local directories of another executor which runs on the same host.
+   */
+  private def readDiskBlockFromSameHostExecutor(
+      blockId: BlockId,
+      localDirs: Array[String],
+      blockSize: Long): Option[ManagedBuffer] = {
+    val file = ExecutorDiskReader.getFile(localDirs, subDirsPerLocalDir, blockId.name)
+    if (file.exists()) {
+      val mangedBuffer = securityManager.getIOEncryptionKey() match {
+        case Some(key) =>
+          // Encrypted blocks cannot be memory mapped; return a special object that does decryption
+          // and provides InputStream / FileRegion implementations for reading the data.
+          new EncryptedManagedBuffer(
+            new EncryptedBlockData(file, blockSize, conf, key))
+
+        case _ =>
+          val transportConf = SparkTransportConf.fromSparkConf(conf, "files")
+          new FileSegmentManagedBuffer(transportConf, file, 0, file.length)
+      }
+      if (blockSize > 0 && mangedBuffer.size() == 0) {
```

Comment:
I'm a little puzzled. `managedBuffer.size()` calls `blockData.size` which just returns the `blockSize` you passed in the constructor above. So it looks like this condition will never be true.